__tests__/src/app/topics/explore/TopicFollowButton.test.tsx

```diff
@@ -22,27 +31,31 @@ vi.mock('@/hooks/user/useUser');
 vi.mock('@/components/ui/use-toast', () => ({
   toast: vi.fn(),
 }));
-vi.mock('@/services/api/axiosInstance');
 vi.mock('swr', () => ({
   mutate: vi.fn(),
 }));
+vi.mock('@/services/api/axiosInstance');
+vi.mocked(followTopicApi);
+vi.mock('@/services/userService', () => ({
+  followTopicApi: vi.fn(),
+}));
+const mockFollowTopicApi = vi.fn();
+vi.mock('@/services/userService', () => ({
+  followTopicApi: mockFollowTopicApi,
+}));
 
-describe('Individual topic follow button', () => {
+describe('Individual Topic Follow Button', () => {
   afterEach(() => {
     cleanup();
     vi.clearAllMocks();
   });
 
   it('should show an error toast when user is not present', () => {
-    (useAuth as unknown as ReturnType<typeof vi.fn>).mockReturnValue({
-      data: null,
-    });
-    (useUser as unknown as ReturnType<typeof vi.fn>).mockReturnValue({
-      user: null,
-    });
-
-    render(<TopicFollowButton />);
+    (useAuth as jest.Mock).mockReturnValue({ data: null });
+    (useUser as jest.Mock).mockReturnValue({ user: null });
```

Comment:
remove the type assertion. It makes the test unreadable and test files are not type checked